LCM/dsc/engine/ConfigurationManager/LocalConfigurationManager.c

```diff
@@ -63,11 +63,72 @@ static LCMTaskNode * g_TaskHead = NULL;
 static pthread_mutex_t g_TaskQueueMutex;
 static int g_TaskQueueShutdown = 0;
 static int g_TaskQueueExists = 0;
+static MI_Char g_CurrentRunningMethodName[LCM_MAX_PATH] = {0};
+
+MI_Boolean TaskAllowedToAddInQueue(LCMTaskNode * node)
+{
+    LCMTaskNode* current;
+    MI_Boolean result = MI_TRUE;
+    pthread_mutex_lock(&g_TaskQueueMutex);
+
+    Context_Invoke_Basic *nodeArgs = (Context_Invoke_Basic *)node->params;
+    if(nodeArgs == NULL || nodeArgs->methodName == NULL)
+    {
+        goto Cleanup;
+    }
+
+    // if incoming request is consistency request.
+    if(Tcscasecmp(nodeArgs->methodName, MSFT_DSCLocalConfigManager_PerformRequiredConfigurationChecks) == 0)
+    {
+        // verify if consistency operation is already queued.
+        current = g_TaskHead;
+        while (current != NULL)
+        {
+            Context_Invoke_Basic *currentNodeArgs = current->params;
+            if(currentNodeArgs == NULL || currentNodeArgs->methodName == NULL)
```

Comment:
Curiosity question - When can current->params be null? What kind of item exists in the queue with methodName == NULL?

Reply:
Currently all the methods are passing param values and methodname is always present.
I added null check so that if somebody makes mistake in future (doesnt add param values), this code will not crash.

---
In reply to: [138801999](https://github.com/Microsoft/PowerShell-DSC-for-Linux/pull/323#discussion_r138801999) [](ancestors = 138801999)